### ISSUE TITLE
Got panic in running 'telepresence intercept'

### ISSUE THREAD

**Issue by JinVei** [telepresenceio/telepresence#2880]
**Describe the bug**

I got a Panic while running below command
```
telepresence intercept hello-world --port 8000
```
Command output error:
```
telepresence: error: connector.CreateIntercept: PANIC: runtime error: invalid memory address or nil pointer dereference

See logs for details (2 errors found): "/Users/xxxx/Library/Logs/telepresence/connector.log"
If you think you have encountered a bug, please run `telepresence gather-logs` and attach the telepresence_logs.zip to your github issue or create a new one: https://github.com/telepresenceio/telepresence/issues/new?template=Bug_report.md .
```

And saw some logs from connector.log prompt that it panic in ValidatePort
```
2022-11-11 16:29:29.8919 error   connector/server-grpc/conn=63/RunCommand-118 : PANIC: runtime error: invalid memory address or nil pointer dereference
github.com/telepresenceio/telepresence/v2/pkg/client/userd.callRecovery
	github.com/telepresenceio/telepresence/v2/pkg/client/userd/grpc.go:51
github.com/telepresenceio/telepresence/v2/pkg/client/userd.(*Service).withSession.func1.1
	github.com/telepresenceio/telepresence/v2/pkg/client/userd/grpc.go:106
runtime.gopanic
	runtime/panic.go:884
go.opentelemetry.io/otel/sdk/trace.(*recordingSpan).End.func1
	go.opentelemetry.io/otel/sdk@v1.10.0/trace/span.go:393
go.opentelemetry.io/otel/sdk/trace.(*recordingSpan).End
	go.opentelemetry.io/otel/sdk@v1.10.0/trace/span.go:432
runtime.gopanic
	runtime/panic.go:884
runtime.panicmem
	runtime/panic.go:260
runtime.sigpanic
	runtime/signal_unix.go:835
github.com/telepresenceio/telepresence/v2/pkg/agentconfig.ValidatePort
	github.com/telepresenceio/telepresence/v2/pkg/agentconfig/portidentifier.go:24
github.com/telepresenceio/telepresence/v2/pkg/agentconfig.NewPortIdentifier
	github.com/telepresenceio/telepresence/v2/pkg/agentconfig/portidentifier.go:34
github.com/telepresenceio/telepresence/v2/pkg/client/userd/trafficmgr.(*serviceProps).portIdentifier
	github.com/telepresenceio/telepresence/v2/pkg/client/userd/trafficmgr/intercept.go:465
github.com/telepresenceio/telepresence/v2/pkg/client/userd/trafficmgr.(*TrafficManager).AddIntercept
	github.com/telepresenceio/telepresence/v2/pkg/client/userd/trafficmgr/intercept.go:681
github.com/telepresenceio/telepresence/v2/pkg/client/userd.(*Service).CreateIntercept.func2
	github.com/telepresenceio/telepresence/v2/pkg/client/userd/grpc.go:281
github.com/telepresenceio/telepresence/v2/pkg/client/userd.(*Service).withSession.func1
	github.com/telepresenceio/telepresence/v2/pkg/client/userd/grpc.go:111
github.com/telepresenceio/telepresence/v2/pkg/client/userd.(*Service).logCall
	github.com/telepresenceio/telepresence/v2/pkg/client/userd/grpc.go:82
github.com/telepresenceio/telepresence/v2/pkg/client/userd.(*Service).withSession
	github.com/telepresenceio/telepresence/v2/pkg/client/userd/grpc.go:90
github.com/telepresenceio/telepresence/v2/pkg/client/userd.(*Service).CreateIntercept
	github.com/telepresenceio/telepresence/v2/pkg/client/userd/grpc.go:278
github.com/telepresenceio/telepresence/v2/pkg/client/userd/commands.(*interceptState).EnsureState
	github.com/telepresenceio/telepresence/v2/pkg/client/userd/commands/intercept.go:825
github.com/telepresenceio/telepresence/v2/pkg/client.WithEnsuredState
	github.com/telepresenceio/telepresence/v2/pkg/client/ensured_state.go:44
github.com/telepresenceio/telepresence/v2/pkg/client/userd/commands.(*interceptCommand).intercept
	github.com/telepresenceio/telepresence/v2/pkg/client/userd/commands/intercept.go:314
github.com/telepresenceio/telepresence/v2/pkg/client/userd/commands.(*interceptCommand).init.func2
	github.com/telepresenceio/telepresence/v2/pkg/client/userd/commands/intercept.go:213
github.com/telepresenceio/telepresence/v2/pkg/client/cli/output.(*output).runE.func1
	github.com/telepresenceio/telepresence/v2/pkg/client/cli/output/output.go:91
github.com/spf13/cobra.(*Command).execute
	github.com/spf13/cobra@v1.5.0/command.go:872
github.com/spf13/cobra.(*Command).ExecuteC
	github.com/spf13/cobra@v1.5.0/command.go:990
github.com/spf13/cobra.(*Command).ExecuteC
	github.com/spf13/cobra@v1.5.0/command.go:938
github.com/spf13/cobra.(*Command).Execute
	github.com/spf13/cobra@v1.5.0/command.go:918
github.com/spf13/cobra.(*Command).ExecuteContext
	github.com/spf13/cobra@v1.5.0/command.go:911
github.com/telepresenceio/telepresence/v2/pkg/client/userd.(*Service).executeCmd
	github.com/telepresenceio/telepresence/v2/pkg/client/userd/grpc.go:657
github.com/telepresenceio/telepresence/v2/pkg/client/userd.(*Service).RunCommand.func1.3
	github.com/telepresenceio/telepresence/v2/pkg/client/userd/grpc.go:751
github.com/telepresenceio/telepresence/v2/pkg/client/userd.(*Service).withSession.func1
	github.com/telepresenceio/telepresence/v2/pkg/client/userd/grpc.go:111
github.com/telepresenceio/telepresence/v2/pkg/client/userd.(*Service).logCall
	github.com/telepresenceio/telepresence/v2/pkg/client/userd/grpc.go:82
```

Seem like that the Panic cause by making a call to the nil 'err'
at https://github.com/telepresenceio/telepresence/blob/v2.8.5/pkg/agentconfig/portidentifier.go#L24

Because 'err' had been set to nil before
at https://github.com/telepresenceio/telepresence/blob/v2.8.5/pkg/agentconfig/portidentifier.go#L22

**Versions (please complete the following information):**
 - Output of `telepresence version`
```
Client: v2.8.5 (api v3)
Root Daemon: v2.8.5 (api v3)
User Daemon: v2.8.5 (api v3)
```
 - Operating system of workstation running `telepresence` commands
```
Macos M1
```
 - Kubernetes environment and Version [e.g. Minikube, bare metal, Google Kubernetes Engine]
```
openshift

kubectl version

Client Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.5", GitCommit:"5c99e2ac2ff9a3c549d9ca665e7bc05a3e18f07e", GitTreeState:"clean", BuildDate:"2021-12-16T08:38:33Z", GoVersion:"go1.16.12", Compiler:"gc", Platform:"darwin/arm64"}
Server Version: version.Info{Major:"1", Minor:"21+", GitVersion:"v1.21.2-1505+6438632813d726-dirty", GitCommit:"6438632813d72669a22c860cd6c218fd4fe4ec24", GitTreeState:"dirty", BuildDate:"2021-10-15T20:20:45Z", GoVersion:"go1.16.6", Compiler:"gc", Platform:"linux/amd64"}
```

**Additional context**
[telepresence_logs.zip](https://github.com/telepresenceio/telepresence/files/9988936/telepresence_logs.zip)

**Comment by thallgren** [comment]
Thanks for reporting this, and yes, I believe you've found what the problem is. I will be fixed in the next release.

It would be interesting to see the port definition of the service that triggered this. It shouldn't really happen because the Kubernetes port names should be valid. Apparently here's a case where that's not true.

**Comment by thallgren** [comment]
Fixed in b00a4316830b1223afefcf9719e05dac42ce3bbf. Will be released with 2.9.0.

**Comment by JinVei** [comment]
The definition of this service like below:
```
apiVersion: v1
kind: Service
metadata:
  annotations:
    openshift.io/generated-by: OpenShiftNewApp
  labels:
    app: hello-world
    app.kubernetes.io/component: hello-world
    app.kubernetes.io/instance: hello-world
  name: hello-world
  namespace: devops
spec:
  clusterIP: 10.31.95.32
  clusterIPs:
  - 10.31.95.32
  ipFamilies:
  - IPv4
  ipFamilyPolicy: SingleStack
  ports:
  - name: 8000-tcp
    port: 8000
    protocol: TCP
    targetPort: http
  selector:
    deployment: hello-world
  sessionAffinity: None
  type: ClusterIP
status:
  loadBalancer: {}
```
After change the name of this port to 'tcp', it  work fine.
Is the port name can not be started with number?
